app/src/main/java/org/mozilla/fenix/HomeActivity.kt

```diff
@@ -1016,18 +1016,22 @@ open class HomeActivity : LocaleAwareAppCompatActivity(), NavHostActivity {
         }
     }
 
+    fun processIntent(intent: Intent): Boolean {
```

Comment:
nit: looks like this could be `private`.

Reply:
(Ah you refactored and added tests 🎉 , maybe a `VisibleForTesting`) :)

Reply:
I made it public intentionally to be able to use the same behavior https://github.com/mozilla-mobile/fenix/pull/24225/files#diff-8e4963255d11a17aaae4c9eb198b24188b6dd43122a75a5fdf8b61e8b38f7457R55